include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -88,6 +89,13 @@ class IExecutionProvider {
     return nullptr;
   }
 
+  /**
+   * Returns an external data loader object that implements methods to load data from external sources.
```

Comment:
Might want to mention that this is optional to implement and it's only useful in specialized scenarios because in all other cases the framework takes care of loading the external data and copying it to the device.